src/Illuminate/Database/Eloquent/Relations/Concerns/InteractsWithPivotTable.php

```diff
@@ -475,7 +475,7 @@ protected function detachUsingCustomClass($ids)
     protected function getCurrentlyAttachedPivots()
     {
         return $this->newPivotQuery()->get()->map(function ($record) {
-            $class = $this->using ? $this->using : Pivot::class;
+            $class = $this->using ?: Pivot::class;
```

Comment:
The first situation I mentioned is valid here.

$this->using It contains 3 rules.

1. It should not be empty.
1. It must send data.
2. If it is empty or not sending data, option 3 is used.

With your action, the php compiler may deliver incorrect results in some cases.